### PR TITLE
Increase dependency

### PR DESCRIPTION
**Pull request by rakshith91**

### CODE REVIEW

File: sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/__init__.py [hunk 1]
```diff
@@ -125,6 +125,11 @@ def kind(self):
     def kind(self, value):
         # type: (SpanKind) -> None
         """Set the span kind of this span."""
+        warnings.warn(
+            """Kind must be set while creating the span for Opentelemetry. It might be possible
+            that one of the packages you are using doesn't follow the latest Opentelemtry Spec.
+            Try updating the azure packages to the latest versions."""
+        )
```
Comment (rakshith91) [inline sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/__init__.py:132]:
```suggestion
```

File: sdk/core/azure-core/azure/core/tracing/_abstract_span.py [hunk 2]
```diff
@@ -221,6 +221,7 @@ def set_http_attributes(self, request, response=None):
         :param response: The response received by the server. Is None if no response received.
         :type response: ~azure.core.pipeline.transport.HttpResponse or ~azure.core.pipeline.transport.AsyncHttpResponse
         """
+        self.kind = SpanKind.CLIENT
```
Comment (lmazuel) [inline sdk/core/azure-core/azure/core/tracing/_abstract_span.py:224]: You're changing the behavior of the mixin, why?

Comment (lmazuel) [reply sdk/core/azure-core/azure/core/tracing/_abstract_span.py:224]: Discussed offline, this is revert of:
https://github.com/Azure/azure-sdk-for-python/pull/17785/files#diff-363cd4d434eb6553b982b1473cd363d4afa4b14c36acec10f060a874f63cbe09L224

Comment (rakshith91) [reply sdk/core/azure-core/azure/core/tracing/_abstract_span.py:224]: actually - i already added the kind in the distributed tracing policy and removed it from here previously in this PR
https://github.com/Azure/azure-sdk-for-python/pull/17785
- duplicating this line here ensures backward compatibility with previosuly released versions of azure-core-tracing-opentelemetry

File: sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/__init__.py [hunk 3]
```diff
@@ -299,3 +299,24 @@ def call_with_current_context(*args, **kwargs):
...
+        self.add_attribute(self._SPAN_COMPONENT, "http")
+        self.add_attribute(self._HTTP_METHOD, request.method)
+        self.add_attribute(self._HTTP_URL, request.url)
+        user_agent = request.headers.get("User-Agent")
+        if user_agent:
+            self.add_attribute(self._HTTP_USER_AGENT, user_agent)
+        if response and response.status_code:
+            self.add_attribute(self._HTTP_STATUS_CODE, response.status_code)
+        else:
+            self.add_attribute(self._HTTP_STATUS_CODE, 504)
```
Comment (rakshith91) [inline sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/__init__.py:322]:
```suggestion
```

File: shared_requirements.txt [hunk 4]
```diff
@@ -127,7 +127,7 @@ pyjwt>=1.7.1
 #override azure azure-keyvault~=1.0
 #override azure-mgmt-core azure-core<2.0.0,>=1.13.0
 #override azure-containerregistry azure-core>=1.4.0,<2.0.0
-#override azure-core-tracing-opencensus azure-core<2.0.0,>=1.0.0
+#override azure-core-tracing-opencensus azure-core<2.0.0,>=1.13.0
```
Comment (rakshith91) [inline shared_requirements.txt:130]:
```suggestion
#override azure-core-tracing-opencensus azure-core<2.0.0,>=1.0.0
```

File: sdk/core/azure-core-tracing-opencensus/setup.py [hunk 5]
```diff
@@ -61,7 +61,7 @@
         'opencensus>=0.6.0',
         'opencensus-ext-azure>=0.3.1',
         'opencensus-ext-threading',
-        'azure-core<2.0.0,>=1.0.0',
+        'azure-core<2.0.0,>=1.13.0',
```
Comment (rakshith91) [inline sdk/core/azure-core-tracing-opencensus/setup.py:64]:
```suggestion
        'azure-core<2.0.0,>=1.0.0',
```

**Comment by rakshith91** [comment]
running the regression tests here https://dev.azure.com/azure-sdk/internal/_build/results?buildId=875505&view=results

**Review by lmazuel — APPROVED** [review]

**Pull request merged**